src/sharing/SharingDialog.component.js

```diff
@@ -1,44 +1,251 @@
+    };
+}
+
+/**
+ * A pop-up dialog for changing sharing preferences for a sharable object.
+ */
+class SharingDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            apiObject: null,
+            objectToShare: null,
+        };
+
+        this.loadObjectFromApi();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.id !== this.props.id) {
+            this.setState({
+                objectToShare: null,
+            });
+
+            this.loadObjectFromApi();
+        }
+    }
+
+    onSearchRequest(searchText) {
+        const apiInstance = this.state.api;
+
+        return apiInstance.get('sharing/search', { key: searchText })
+            .then((searchResult) => {
+                let transformedResult = searchResult.users.map(
+                    user => transformAccessObject(user, 'user'));
+
+                transformedResult = transformedResult.concat(
+                    searchResult.userGroups.map(
+                        userGroup => transformAccessObject(userGroup, 'userGroup')));
+
+                return transformedResult;
+            });
+    }
+
+    onSharingChanged(updatedAttributes, onSuccess) {
+        const objectToShare = {
+            ...this.state.objectToShare,
+            ...updatedAttributes,
+        };
+
+        const apiObject = this.restoreObjectStructure(objectToShare);
+
+        return this.state.api.post(`sharing?type=${this.props.type}&id=${this.props.id}`, apiObject)
+          .then(({ httpStatus, message }) => {
+              if (httpStatus === 'OK') {
+                  this.setState({
+                      objectToShare,
+                      apiObject,
+                  }, () => {
+                      onSuccess && onSuccess();
+                  });
+              } else {
+                  console.warn('Failed to post changes.');
+                  console.warn('SERVER SAID:', message);
+              }
 
-    mixins: [Translate],
+              return message;
+          })
+          .catch(({ message }) => {
+              console.warn('Failed to post changes.');
+              console.warn('SERVER SAID:', message);
+          });
+    }
+
+    loadObjectFromApi() {
+        getInstance().then((d2) => {
+            const apiInstance = d2.Api.getApi();
+            apiInstance.get('sharing', { type: this.props.type, id: this.props.id })
+                .then((apiObject) => {
+                    this.setState({
+                        api: apiInstance,
+                        apiObject,
+                        objectToShare: transformObjectStructure(apiObject.meta, apiObject.object),
+                    });
+                });
+        });
+    }
+
+    restoreObjectStructure(transformedObject) {
+        const userAccesses = [];
+        const userGroupAccesses = [];
+
+        // eslint-disable-next-line array-callback-return
+        transformedObject.accesses.map((access) => {
```

Comment:
If you're not going to use the result of the `.map` you can use `.forEach` (which would also get rid of the eslint error ;))